adapters/telegram/src/polling.ts

```diff
@@ -29,13 +30,30 @@ export class HttpPolling extends Adapter.Client<TelegramBot> {
           return bot.offline()
         }
 
-        const updates = await bot.internal.getUpdates({
-          offset: this.offset + 1,
-          timeout: bot.config.pollingTimeout,
-        })
-        for (const e of updates) {
-          this.offset = Math.max(this.offset, e.update_id)
-          handleUpdate(e, bot)
+        try {
+          const updates = await bot.internal.getUpdates({
+            offset: this.offset + 1,
+            timeout: bot.config.pollingTimeout,
+          })
+          if (bot.status === 'reconnect') bot.status = 'online'
```

Comment:
Maybe there should be `bot.online()`. Status should be set to `online` in other situations as well.